### ISSUE TITLE
bump github.com/moby/buildkit from 0.11.6 to 0.12.4 (requires some manual massaging it seems)

### ISSUE THREAD

**Issue by aaron-prindle** [GoogleContainerTools/kaniko#2954]
Kaniko should update github.com/moby/buildkit from 0.11.6 to 0.12.4.  When this is attempted simply using multiple `go get` statements and trying to update multiple transitive deps, there is always the same compilation error:

`log snippet`
```
Error: vendor/github.com/moby/buildkit/frontend/dockerfile/instructions/parse.go:592:15: healthcheck.StartInterval undefined (type container.HealthConfig has no field or method StartInterval)
```
`full unit test logs w/ error from dependabot PR bumping the deps`
error logs from dependabot PR attempting to update github.com/moby/buildkit from 0.11.6 to 0.12.4:
https://github.com/GoogleContainerTools/kaniko/actions/runs/7551895883/job/20559869275?pr=2925

**Comment by jjbustamante** [comment]
From what I can understand the problem is:

- This [code](https://github.com/GoogleContainerTools/kaniko/blob/main/pkg/commands/healthcheck.go#L31-L36) is trying to cast a docker *HealthCheck* structure to a ggcr *HealthCheck* structure.

**Comment by amascia-gg** [comment]
Hi, any updates on this? I'm facing the same issue and I need version `v0.12.5` of `github.com/moby/buildkit` because it fixes a lot of [CVEs](https://osv.dev/list?ecosystem=Go&q=buildkit).

**Comment by aaron-prindle** [comment]
@amascia-gg, this is done now via https://github.com/GoogleContainerTools/kaniko/pull/2995  Using the latest version of Kaniko - `v1.20.1`, the `executor` image has no fixable CVEs:

```
aprindle@aprindle-ssd ~/kaniko  [main]grype gcr.io/kaniko-project/executor:v1.20.1
 ✔ Vulnerability DB        [updated]
New version of grype is available: 0.74.5 (currently running: 0.61.0)
 ✔ Loaded image            
 ✔ Parsed image            
 ✔ Cataloged packages      [211 packages]
 ✔ Scanning image...       [2 vulnerabilities]
   ├── 0 critical, 1 high, 1 medium, 0 low, 0 negligible
   └── 0 fixed

```

Closing
